### PR TITLE
Fix Github Actions failure

### PR DESCRIPTION
**Pull request by ihor-sviziev**
# Description (*)

The last Github Actions ran showed that it started failing:
https://github.com/magento/magento-semver/pull/61/checks
image

This PR changes how PHP is getting installed to use https://github.com/marketplace/actions/setup-php-action for installing correct PHP version.

## Fixed Issues (if relevant)

1. magento/magento-semver#<issue_number>: Issue title

## Manual testing scenarios (*)

1. ...

2. ...

## Contribution checklist (*)

- [ ] Pull request has a meaningful description of its purpose
- [ ] All commits are accompanied by meaningful commit messages
- [ ] All new or changed code is covered with unit/integration tests (if applicable)
- [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -25,13 +33,13 @@ jobs:
...
-        key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+        key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
         restore-keys: |
-          ${{ runner.os }}-php-
+          ${{ runner.os }}-composer-
 
     - name: Install dependencies
       if: steps.composer-cache.outputs.cache-hit != 'true'
-      run: composer install --prefer-source --no-interaction
+      run: composer install --prefer-dist --no-interaction
```
Comment (fascinosum) [inline .github/workflows/php.yml:42]: this is the default behavior. I think we can omit it

File: .github/workflows/php.yml [hunk 2]
```diff
@@ -10,12 +10,20 @@ jobs:
   build:
 
     runs-on: ubuntu-latest
-
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions: ['7.2', '7.3', '7.4']
```
Comment (fascinosum) [inline .github/workflows/php.yml:16]: Do we even need to test it on PHP 7.2 if it reached the end of life almost year ago?
I would even recommend removing PHP 7.2 support from composer.json
cc: @vkublytskyi

**Review by fascinosum — CHANGES_REQUESTED** [review]

File: .github/workflows/php.yml (hunk 1 above)
Comment (ihor-sviziev) [reply .github/workflows/php.yml:42]: Could you explain, what is the default behavior?

File: .github/workflows/php.yml (hunk 2 above)
Comment (ihor-sviziev) [reply .github/workflows/php.yml:16]: I'm ok with it, but currently, it is there, so it's better to make sure it is working with all declared versions

File: .github/workflows/php.yml (hunk 1 above)
Comment (ihor-sviziev) [reply .github/workflows/php.yml:42]: Btw, maybe it's better to switch to testing on the lowest and highest dependencies? What do you think?

File: .github/workflows/php.yml (hunk 2 above)
Comment (ihor-sviziev) [reply .github/workflows/php.yml]: I just added commit to remove PHP 7.2 support

File: .github/workflows/php.yml (hunk 1 above)
Comment (ihor-sviziev) [reply .github/workflows/php.yml]: Switched to https://github.com/marketplace/actions/install-composer-dependencies, it supports also running on the lowest & highest dependencies

**Comment by ihor-sviziev** [comment]
@fascinosum, could you, please review this PR as well?

**Review by slavvka — APPROVED** [review]

**Review by fascinosum — APPROVED** [review]

**Pull request merged**